### ISSUE TITLE
False positive on `Style/RedundantLineContinuation`

### ISSUE THREAD

**Issue by shinnc** [rubocop/rubocop#12549]
## Expected behavior

```ruby
describe Test do
  it do
    x = 1
    y = 2

    expect { described_class.test(x, y) }
      .to change { x }
      .from(1)
      .to(2) \ # No offence

      .and change { y }
      .from(2)
      .to(3)
  end
end
```

## Actual behavior

Rubocop complains we don't need `\`:

```bash
Inspecting 1 file
C

Offenses:

test.rb:13:14: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
      .to(2) \ ...
             ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

However with `\` removed (no longer valid Ruby syntax), running rubocop raise:

```bash
Inspecting 1 file
F

Offenses:

test.rb:15:7: F: Lint/Syntax: unexpected token tDOT
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
      .and change { y }
      ^
test.rb:19:1: F: Lint/Syntax: unexpected token kEND
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^

1 file inspected, 2 offenses detected
```

## Steps to reproduce the problem

```bash
cat << EOF > test.rb
class Test
  def self.test(x, y); end
end

describe Test do
  it do
    x = 1
    y = 2

    expect { described_class.test(x, y) }
      .to change { x }
      .from(1)
      .to(2) \\

      .and change { y }
      .from(2)
      .to(3)
  end
end
EOF

bundle exec rubocop test.rb --only Style/RedundantLineContinuation
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin23]
  - rubocop-performance 1.20.0
  - rubocop-rails 2.23.0
  - rubocop-rspec 2.25.0
```
